### PR TITLE
Improve formulation of rational clock.

### LINKED ISSUE #2386: Rational Clock Constructors with variable intervalCounter argument
Currently the specification reads 

> The first input argument, intervalCounter, is a clocked Component Expression (see Section 16.2.3) or a parameter expression of type Integer with min=0. The optional second argument resolution (default=1) is a parameter expression of type Integer with min=1 and unit “Hz”. If intervalCounter is a parameter expression with value zero, the period of the clock is derived by clock inference, see Section 16.7.5.

The unclear part is the last sentence
> If intervalCounter is a **parameter** expression with value zero, the period of the clock is derived by clock inference, see Section 16.7.5.

What happens if we have a clocked component expression that evaluates to zero, e.g
```
model ClockConstructorRationalIntervalInferenceInvalid
	Integer intervalCounter = 0;
	Clock rationalClock = Clock(intervalCounter); // Only parameter expressions are allowed for clock inference
end ClockConstructorRationalIntervalInferenceInvalid;
```

I dont see any compelling reason why it should be invalid. More importantly one could build systems with variable interval where the frequency is driven by one component, e.g.
```
model MultipleClocksVariableInterval
	Integer frequency1(start = 2);
	Integer frequency2(start = 0);
	Clock baseClock1 = Clock(frequency1, 10);
	Clock baseClock2 = Clock(frequency2, 10);
        Boolean someCondition;
	equation
                when sample(someCondition, baseClock1) then
                     frequency1 = 0;
                end when;
                when sample(someCondition, baseClock2) then
                     frequency2 = 2;
                end when;
end MultipleClocksVariableInterval;
```

This looks like it would be a valid use case. Consequently I would propose that we rephrase the last sentence to:

> If intervalCounter evaluates to zero, the period of the clock is derived by clock inference, see Section 16.7.5.

Note that this would also imply that we have to perform clock inference during run time. However this is not a realy problem, as every factor from the subclock conversion operators is parametric and needs to be determined only once. Consequently the "runtime clock inference" would be quite cheap as we would only have to check for consistence.

### PR DESCRIPTION
**Pull request by HansOlsson**
It might be that we should have a clearer separator for the =0 case.
Closes #2386

### CODE REVIEW

File: chapters/synchronous.tex [hunk 1]
```diff
@@ -342,17 +342,18 @@ \section{Clock Constructors}\label{clock-constructors}
 \end{lstlisting}\end{synopsis}
 \begin{semantics}
 \firstuse{Rational interval clock}.
-The first input argument, $\mathit{intervalCounter}$, is a clocked component expression (\cref{def:component-expression}) or a parameter expression of type \lstinline!Integer! with \lstinline!min = 0!.
+The first input argument, $\mathit{intervalCounter}$, is a clocked component expression (\cref{def:component-expression}), or a parameter expression of type \lstinline!Integer! with \lstinline!min = 0!.
 The optional second argument $\mathit{resolution}$ (defaults to 1) is a parameter expression of type \lstinline!Integer! with \lstinline!min = 1! and \lstinline!unit = "Hz"!.
 If $\mathit{intervalCounter}$ is a parameter expression with value zero, the period of the clock is derived by clock inference, see \cref{sub-clock-inferencing}.
+
+If $\mathit{intervalCounter}$ is a parameter expression greater than zero, the clock defines a periodic clock.
+If $\mathit{intervalCounter}$ is a clocked component expression it must greater than zero.
```
Comment (eshmoylova) [inline chapters/synchronous.tex:350]: Missing "be" in "it must greater than zero."

File: chapters/synchronous.tex [hunk 2]
```diff
@@ -342,17 +342,18 @@ \section{Clock Constructors}\label{clock-constructors}
 \end{lstlisting}\end{synopsis}
 \begin{semantics}
 \firstuse{Rational interval clock}.
-The first input argument, $\mathit{intervalCounter}$, is a clocked component expression (\cref{def:component-expression}) or a parameter expression of type \lstinline!Integer! with \lstinline!min = 0!.
+The first input argument, $\mathit{intervalCounter}$, is a clocked component expression (\cref{def:component-expression}), or a parameter expression of type \lstinline!Integer! with \lstinline!min = 0!.
```
Comment (eshmoylova) [inline chapters/synchronous.tex:345]: I feel like the comma is not necessary here. There are only two items  in the list. Usually, it is used for in lists of three or more items.

File: chapters/synchronous.tex (hunk 1 above)
Comment (HansOlsson) [reply chapters/synchronous.tex]: Ok, added.

File: chapters/synchronous.tex (hunk 2 above)
Comment (HansOlsson) [reply chapters/synchronous.tex]: Ok, removed.

**Review by eshmoylova — APPROVED** [review]

**Pull request merged**